gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -510,13 +510,18 @@ public void handleLaunchFlowEvent(DagActionStore.DagAction launchAction) {
       this.dagActionStore.get().deleteDagAction(launchAction);
     } catch (URISyntaxException e) {
       log.warn("Could not create URI object for flowId {} due to exception {}", flowId, e.getMessage());
+      this.dagManagerMetrics.incrementFailedLaunchCount();
     } catch (SpecNotFoundException e) {
       log.warn("Spec not found for flowId {} due to exception {}", flowId, e.getMessage());
+      this.dagManagerMetrics.incrementFailedLaunchCount();
     } catch (IOException e) {
       log.warn("Failed to add Job Execution Plan for flowId {} OR delete dag action from dagActionStore (check "
           + "stacktrace) due to exception {}", flowId, e.getMessage());
+      this.dagManagerMetrics.incrementFailedLaunchCount();
     } catch (InterruptedException e) {
-      log.warn("SpecCompiler failed to reach healthy state before compilation of flowId {}. Exception: ", flowId, e);
+      log.warn("SpecCompiler failed to reach healthy state before compilation of flowId {} due to exception {}", flowId,
+          e);
```

Comment:
I guess there's the possibility I may be misremembering... and pursuing this needlessly... but my expectation is that a stacktrace would only be written when calling this form:
```
Logger::warn(String, Throwable)
```
if you call the form:
```
Logger::warn(String, Object, Object) // aka. Logger::warn(String, Object...)
```
are you certain it will print the ST, when the last arg is `Throwable` and there's no corresponding `{}` remaining for it in the initial `String` arg?

Reply:
There are two `{}` and the last one is for the throwable so it will print the `getMessage`. I updated all the methods to use `Logger::warn(String, Throwable)` thought to get full stack trace.